tyr-runner/src/main/java/org/jboss/tyr/ci/TeamCityCI.java

```diff
@@ -73,57 +110,103 @@ public void triggerBuild(JsonObject prPayload) {
         String sha = prPayload.getJsonObject(Utils.HEAD).getString(Utils.SHA);
         String buildId = branchMappings.get(branch);
 
-        Client client = ClientBuilder.newClient();
-        URI statusUri = UriBuilder
-                .fromUri(baseUrl)
-                .path("/app/rest/buildQueue")
-                .build();
+        setSnapshotDependencies(buildId, retestOnlyFailed);
 
-        WebTarget target = client.target(statusUri);
+        String path = "/app/rest/buildQueue";
+        WebTarget target = getTeamCityTarget(path);
         Entity<BuildJson> json = Entity.json(new BuildJson("pull/" + pull, buildId, sha, pull, branch));
 
         Response response = null;
 
         try {
-            response = target.request()
-                .header(HttpHeaders.ACCEPT_ENCODING, "UTF-8")
-                .header(HttpHeaders.CONTENT_TYPE, MediaType.APPLICATION_JSON)
-                .header(HttpHeaders.AUTHORIZATION, "Basic " + encryptedCredentials)
-                .post(json);
-
+            response = getHeaderBuilder(target).post(json);
             log.info("Teamcity status update: " + response.getStatus());
         } catch (Throwable e) {
             log.error("Cannot run Team city build", e);
         } finally {
             if (response != null) {
                 response.close();
+                client.close();
             }
         }
     }
 
+    private void setSnapshotDependencies(String buildId, boolean retestOnlyFailed) {
+        SnapshotDependencies snapshotDependencies = getSnapshotDependencies(buildId);
 
-    private String encryptCredentials(String username, String password) {
-        String authStr = username + ":" + password;
-        return DatatypeConverter.printBase64Binary(authStr.getBytes(StandardCharsets.UTF_8));
-    }
+        if (snapshotDependencies.snapshotDependencyCount.intValue() < 1) {
+            log.info("No snapshot dependency.");
+            return;
+        }
 
-    private String getBaseUrl(String host, int port) {
-        return String.format("http://%s:%d/httpAuth", host, port);
+        for (SnapshotDependency dependency: snapshotDependencies.snapshotDependencies) {
+            for (Property prop :dependency.properties.propertyList) {
```

Comment:
just minor code style preference, use space around ':' on both sides in for-each loops please